pkg/process/io.go

```diff
@@ -254,86 +261,133 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (runc.IO, error)
 			args = append(args, vs[0])
 		}
 	}
-	ctx, cancel := context.WithCancel(ctx)
-	cmd := exec.CommandContext(ctx, uri.Path, args...)
-	cmd.Env = append(cmd.Env,
-		"CONTAINER_ID="+id,
-		"CONTAINER_NAMESPACE="+ns,
-	)
+
 	out, err := newPipe()
 	if err != nil {
-		cancel()
 		return nil, err
 	}
+
 	serr, err := newPipe()
```

Comment:
we should close pipe if meet error. since it is not related to this issue, it can be follow-up pr.

Reply:
Will follow up.